src/TokenizedStrategy.sol

```diff
@@ -1144,40 +1144,51 @@ contract TokenizedStrategy {
      * A report() call will be needed to record the profit.
      */
     function tend() external nonReentrant onlyKeepers {
+        // Tend the strategy with the current totalIdle.
+        IBaseTokenizedStrategy(address(this)).tendThis(
+            _strategyStorage().totalIdle
+        );
+
+        // Update balances based on ending state.
+        _updateBalances();
+    }
+
+    /**
+     * @notice Update the internal balances that make up `totalAssets`.
+     * @dev This will update the ratio of debt and idle that make up
+     * totalAssets based on the actual current loose amount of `asset`
+     * in a safe way. But will keep `totalAssets` the same, thus having
+     * no effect on Price Per Share.
+     */
+    function _updateBalances() internal {
         StrategyData storage S = _strategyStorage();
```

Comment:
So a minor tradeoff here is we need to get the storage pointer twice?

Reply:
Yes. In order to reuse the same logic.

Though it does do one less asset.balanceOf call during tend that helps offset that extra the gas usage.